docs/site/Extending-OpenAPI-specification.md

```diff
@@ -0,0 +1,154 @@
+---
+lang: en
+title: 'Extending OpenAPI Specification'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Extending-OpenAPI-specification.html
+---
+
+## OpenAPI Specification Enhancer
+
+The APIs in a LoopBack `RestApplication` are described by the
+[OpenAPI Specification(short for OAS)](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md).
+An application's OAS are mainly generated from
+[controllers](https://loopback.io/doc/en/lb4/Controllers.html) and their
+members' metadata, but we would also like to contribute specifications from
+other places. Therefore an extension point `OASEnhancerService` is created to
+allow registered extensions to provide their OAS fragments and modify a rest
+application's specification.
+
+_Read about the extension point/extension patten in
+[documentation](https://loopback.io/doc/en/lb4/Extension-point-and-extensions.html)_
+
+## Adding a New OAS Enhancer
+
+Interface `OASEnhancer` is created in `@loopback/openapi-v3` to describe the
+specification enhancers. A typical OAS enhancer class should have a `name` as
```

Comment:
Suggestion:
`A typical OAS enhancer class should have a 'name' as string, and a ...` ->
`A typical OAS enhancer class should have a string type 'name' field ( no comma) and a function 'modifySpec()' that can modify the current specification`